util/winutil/svcdiag_windows.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package winutil
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+	"tailscale.com/types/logger"
+)
+
+// LogSvcState obtains the state of the Windows service named rootSvcName and
+// all of its dependencies, and then emits that state to logf.
+func LogSvcState(logf logger.Logf, rootSvcName string) {
+	logEntries := []svcStateLogEntry{}
+
+	walkFn := func(svc *mgr.Service, config mgr.Config) {
+		status, err := svc.Query()
+		if err != nil {
+			logf("Failed retrieving Status for service %q: %v", svc.Name, err)
+		}
+
+		logEntries = append(logEntries, makeLogEntry(svc, status, config))
+	}
+
+	err := walkServices(rootSvcName, walkFn)
+	if err != nil {
+		logf("LogSvcState: %v", err)
```

Comment:
```suggestion
		logf("LogSvcState: error: %v", err)
```

Nitpick, so feel free to ignore: you can put `error: ` here and drop the `failed ...` from the `fmt.Errorf` calls below.